File: java/poker/src/main/java/com/duchen/poker/model/type/DoubleTypeChecker.java
package com.duchen.poker.model.type;

import java.util.Arrays;
import java.util.List;

public class DoubleTypeChecker implements TypeChecker {

    @Override
    public CardGroupType getType() {
        return CardGroupType.DOUBLE;
    }

    @Override
    public List<Integer> getMyPossibleCardCount() {
        return Arrays.asList(2);
    }

    @Override
    public int getMaxCardIfIsMyType(List<Integer> cards) {
        if (cards.get(0).intValue() == cards.get(1).intValue()) {
            return cards.get(0);
        }
        return -1;
    }

}

File: java/poker/src/main/kotlin/poker/model/type/DoubleTypeChecker.kt
package com.duchen.poker.model.type

import java.util.Arrays

class DoubleTypeChecker : TypeChecker {

    override val type: CardGroupType
        get() = CardGroupType.DOUBLE

    override val myPossibleCardCount: List<Int>
        get() = Arrays.asList(2)

    override fun getMaxCardIfIsMyType(cards: MutableList<Int>): Int {
        return if (cards[0] == cards[1]) {
            cards[0]
        } else -1
    }

}
